laa_crime_application_store_app/services/v1/application_service.py

```diff
@@ -123,6 +125,9 @@ def update_existing_application(
         ):
             return existing_application.id
 
+        if not get_permissions().allow_update(existing_application, roles):
```

Comment:
We should have some sort of locking here. At the moment, it's not impossible that the provider app would try to update a `sent_back` PA application at the precise moment the caseworker app would try to mark it as expired. We don't want them both to pass the permission check simultaneously and then both apply their updates to the same record.

Reply:
So not going to worry about this in this PR as due to one being an automated task and the other manual submission there is low likelihood of overlap with submission times - if we are worries I would suggest changing the expiry task from 11pm to 4am.

Reply:
Is it not like a 2-line addition to add a 'SELECT FOR UPDATE'? The point of these PRs is to protect data integrity, and I'd rather our strategy not be "hope a lawyer doesn't pull an all-nighter"